csrc/ir/nodes.cpp

```diff
@@ -4881,6 +4881,13 @@ bool ForLoop::isTrivial() const {
     return true;
   }
 
+  if (start()->isConstScalar() && simplifiedStop()->isConstScalar() &&
```

Comment:
Because of the simplification, the pattern previously detected by the above condition, e.g., `for (int i = ceilDiv(32, 8) - 1; i < ceilDiv(32, 8); ++i)` may become `for (int = 3; i < 4; ++i)`, so this condition is necessary to detect that pattern too.